# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/other/RemoteAccessible.cpp
# Commit: 6591f14bbc71
# Full Hash: 6591f14bbc713f11b6316712d931ecc19a443a5c
# Author: Morgan Reschenberg <mreschenberg@mozilla.com>
# Date: 2021-10-05 03:52:32
# Regressor Bug: 1726227
# File Overlap Count: 1
# Description:
#   Bug 1726227: Cache parent-relative accessible bounds, resolution in parent process r=Jamie,eeejay
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D123399
# ==============================================================================

diff -r 06dda9072cb7 -r 6591f14bbc71 accessible/ipc/other/RemoteAccessible.cpp
--- a/accessible/ipc/other/RemoteAccessible.cpp	Mon Oct 04 19:28:49 2021 +0000
+++ b/accessible/ipc/other/RemoteAccessible.cpp	Mon Oct 04 19:46:54 2021 +0000
@@ -15,6 +15,7 @@
 #include "mozilla/a11y/Platform.h"
 #include "RelationType.h"
 #include "mozilla/a11y/Role.h"
+#include "mozilla/StaticPrefs_accessibility.h"
 
 namespace mozilla {
 namespace a11y {
@@ -879,7 +880,11 @@
   return target;
 }
 
-nsIntRect RemoteAccessible::Bounds() {
+nsIntRect RemoteAccessible::Bounds() const {
+  if (StaticPrefs::accessibility_cache_enabled_AtStartup()) {
+    return RemoteAccessibleBase<RemoteAccessible>::Bounds();
+  }
+
   nsIntRect rect;
   Unused << mDoc->SendExtents(mID, false, &(rect.x), &(rect.y), &(rect.width),
                               &(rect.height));